lib/matplotlib/tests/test_sphinxext.py

```diff
@@ -2,14 +2,155 @@
 
 import filecmp
 from os.path import join as pjoin, dirname, isdir
+import pathlib
 from subprocess import Popen, PIPE
 import sys
+import warnings
 
 import pytest
 
-
+# Only run the tests if Sphinx is installed.
 pytest.importorskip('sphinx')
 
+# Docutils is a dependency of Sphinx so it is safe to
+# import after we know Sphinx is available.
+from docutils.nodes import caption, figure               # noqa: E402
+
+# Sphinx has some deprecation warnings we don't want to turn into errors.
+with warnings.catch_warnings():
```

Comment:
What are these warnings? I feel like long term we don't want to just ignore these.

Reply:
It's Jinja (imported by Sphinx) giving a DeprecationWarning: "Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated, and in 3.8 it will stop." This has been fixed in Jinja's master and should be included in the next release; see pallets/jinja#1028. The other warning (in `build_test_doc()` is the same warning, also from Jinja and should also be fixed in the next release.